File: src/Application Specific/Classes/Entity Collections/Jobs/Job/JobInstall/JobInstallValidator.cs
using System.Collections;
using Microsoft.VisualBasic.CompilerServices;

namespace FSM.Entity
{
    namespace JobInstalls
    {
        // make sure that contact object is valid
        public class JobInstallValidator : BaseValidator
        {
            public void Validate(JobInstall oJobInstall)
            {
                // make sure that contact object is valid
                if (oJobInstall.Errors.Count > 0)
                {
                    foreach (DictionaryEntry de in oJobInstall.Errors)
                        AddCriticalMessage(Conversions.ToString(de.Value));
                }

                if (ValidatorMessages.CriticalMessages.Count > 0)
                {
                    throw new ValidationException(this);
                }
            }
        }
    }
}

File: src/Application Specific/Classes/Entity Collections/Jobs/Job/JobInstall/JobInstallValidator.vb
Namespace Entity

    Namespace JobInstalls

        Public Class JobInstallValidator
            'make sure that contact object is valid
            Inherits BaseValidator

            Public Sub Validate(ByVal oJobInstall As JobInstall)

                'make sure that contact object is valid
                If oJobInstall.Errors.Count > 0 Then
                    Dim de As DictionaryEntry
                    For Each de In oJobInstall.Errors
                        Me.AddCriticalMessage(CStr(de.Value))
                    Next
                End If



                If Me.ValidatorMessages.CriticalMessages.Count > 0 Then
                    Throw New ValidationException(Me)
                End If

            End Sub

        End Class

    End Namespace

End Namespace
